src/commands.rs

```diff
@@ -354,6 +354,36 @@ impl AssetCommand {
         patch(&self.authentication, &endpoint, &json!({ "headers": map }))
     }
 
+    pub fn update_web_asset_headers(
+        &self,
+        id: &str,
+        headers: Vec<(String, String)>,
+    ) -> anyhow::Result<(), CommandError> {
+        // if no headers provided update has nothing to do
+        if headers.is_empty() {
+            return Ok(());
+        }
+
+        let mut new_headers: HashMap<_, _> = headers.into_iter().collect();
+        let asset = self.get(id)?;
+        if let Some(assets) = asset.value.as_array() {
+            if assets.is_empty() {
+                return Err(CommandError::MissingField);
+            }
+            let headers = assets[0].get("headers").ok_or(CommandError::MissingField)?;
+            let old_headers = serde_json::from_value::<HashMap<String, String>>(headers.clone())?;
+            debug!("Old headers {:?}", &old_headers);
+            for (key, value) in old_headers {
+                new_headers.entry(key).or_insert(value);
```

Comment:
Hm - do you need to serialize/deserialize old_headers to/from json to mutate? It could be HashMap

Reply:
How else can I iterate over then ? I did not find a way if you know how please tell me

Reply:
HashMap supports iter method:
https://doc.rust-lang.org/std/collections/struct.HashMap.html#method.iter

Reply:
What hashmap? It's serde Value